ksql-engine/src/main/java/io/confluent/ksql/engine/TopicAccessValidatorFactory.java

```diff
@@ -37,9 +37,17 @@ public static TopicAccessValidator create(
       final ServiceContext serviceContext,
       final MetaStore metaStore
   ) {
-    if (isKafkaAuthorizerEnabled(serviceContext.getAdminClient())) {
-      LOG.info("KSQL topic authorization checks enabled.");
-      return new AuthorizationTopicAccessValidator();
+    final AdminClient adminClient = serviceContext.getAdminClient();
+
+    if (isKafkaAuthorizerEnabled(adminClient)) {
+      if (KafkaClusterUtil.isAuthorizedOperationsSupported(adminClient)) {
+        LOG.info("KSQL topic authorization checks enabled.");
+        return new AuthorizationTopicAccessValidator();
+      }
+
+      LOG.info("The Kafka broker has an authorization service enabled, but the Kafka "
```

Comment:
I think we should log this at warning level. The resulting behavior is likely not what the user intended.

Reply:
The topic access validator is used mostly to provide better permission error messages when the KSQL user won't be able to execute a persistent query on KSQL.  I don't think a warning message is a correct level for that case (I am not against about it, though). However, failing KSQL because is not using the correct version of Kafka might be out of scope of our current release. Do we have a compatibility list of Kafka versions that KSQL should support?

Reply:
> Do we have a compatibility list of Kafka versions that KSQL should support?
Not that I'm aware of, no.

I agree with @spena that it's probably a bit late to introduce a code-path to hard-fail here -- though switching the log level up to `warn` does feel like the right play to me. @spena Can you expand a bit about why you think `info` is right here? Maybe I'm missing some nuance.

Reply:
I feel it's not a `warn` because if permission checks are not available, KSQL will still fail further in the code without providing good error messages. However, it is harmless to switch the level to `warn` now that I am thinking about it. I will do the switch and update this PR.